DetoxSync/DetoxSync/SyncManager/DTXSyncManager.m

```diff
@@ -643,6 +644,27 @@ + (void)untrackDisplayLink:(CADisplayLink *)displayLink
 	return [DTXSingleEventSyncResource singleUseSyncResourceWithObjectDescription:objectDescription eventDescription:description];
 }
 
++ (NSDictionary<NSString *, id> *)_syncStatus {
+  NSMutableArray<NSDictionary *> * busyResourcesDescriptions = [NSMutableArray new];
+
+  for(DTXSyncResource* resource in _registeredResources.allObjects) {
```

Comment:
Thread safety: Make sure the iterator is not exposed to runtime changes in `_registeredResources.allObjects`

Reply:
filter method?

Reply:
extracted 👍🏼

Reply:
About thread safety - great point. I verified and indeed everything is dispatched over the same queue so there shouldn't be any runtime changes during the iterations ✔️